### PR TITLE
Doesn't throw when `FormattedMessage` has `defaultMessage`

### PR DESCRIPTION
**Pull request by torifat**
Currently `react-intl` throws if components are not wrapped inside an `IntlProvider`. With this patch if `FormattedMessage` has `defaultMessage` then it won't throw anymore but in development mode it will still show warnings.

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by torifat** [comment]
I was wondering if there's any way I can help expedite the review process :)

File: test/unit/components/message.js
Comment (jnak) [inline test/unit/components/message.js]: It would be clearer to update the description of the test above with "... and there is no defaultMessage".

File: test/unit/components/message.js
Comment (jnak) [inline test/unit/components/message.js]: It would safer to check that "Hello" is properly rendered.

File: src/components/message.js
Comment (jnak) [inline src/components/message.js]: This is going to break with React-Native.
See https://github.com/yahoo/react-intl/wiki/Components#using-react-intl-with-react-native

I wonder if there is simple way to return a <Text /> in RN environment.  @redonkulus do you have any thoughts on that?

**Review by jnak — COMMENTED** [review]
I'm not a usual contributor to react-intl but I would love to help this PR move forward.

If there is no easy workaround the 'span' issue, it may make sense to focus on https://github.com/yahoo/react-intl/pull/1080 instead.

**Comment by jnak** [comment]
> If there is no easy workaround the 'span' issue, it may make sense to focus on #1080 instead.
Actually I'm not sure https://github.com/yahoo/react-intl/pull/1080 actually handles this any better. I would need to write a specific test case for this

File: src/components/message.js
Comment (redonkulus) [reply src/components/message.js]: I'm not entirely comfortable with forking in the library for different environments but I can understand the concern. Unfortunately, I'm not sure if there is an environment variable or flag we can use to conditionally add the default node.

File: test/unit/components/message.js
Comment (torifat) [reply test/unit/components/message.js]: Addressed :)

File: test/unit/components/message.js
Comment (torifat) [reply test/unit/components/message.js]: Addressed :)

**Comment by torifat** [comment]
> it may make sense to focus on #1080 instead
This is a subset of #1080 which will unblock a lot of people I hope. I don't find any reason to ignore one because of the other.

Anyways, I have addressed your comments. And, regarding the `span` I have same opinion as @redonkulus. Currently, it's handled in the same way in others part of the codebase. So, it's not breaking something new.

File: src/components/message.js
Comment (jnak) [reply src/components/message.js]: Actually I don't think we will be able to default to a <RN.Text /> component in React Native without adding an explicit dependency on the "react-native" package. The best thing we can do here is to show a helpful error message in dev.

**Comment by jnak** [comment]
@torifat Thanks for addressing the comments.

Actually I just realized there is no way can default to a <RN.Text /> since we don't depend on the "react-native" package. Given that we are logging a useful error message in dev that will also show up in the RN console, I think it's fine to merge this PR as is. 

@redonkulus Do you have any concerns that you would like to be addressed before merging this?

**Review by redonkulus — APPROVED** [review]
This looks good to me. Thanks for contributing!

**Comment by jnak** [comment]
@redonkulus @torifat Great. Thanks for the quick turnaround!

**Comment by redonkulus** [comment]
This has been released in v2.6.0.

File: src/components/message.js
Comment (redonkulus) [reply src/components/message.js]: @giautm We could return a string, but since this library works with older versions of React, I wouldn't want to break existing applications. If we decide to bump major for the new React Context API, then we could look into making this change here too. Please file an issue outlining the enhancement changes and we can discuss with the community there. Thanks!

File: src/components/message.js
Comment (jzhangnu) [inline src/components/message.js:22]: Anyone could explain this line code for me? Why throw this error when `process.env.NODE_ENV !== 'production'`?
I got stuck by this error in my project. Thanks a lot!

Comment (longlho) [reply src/components/message.js:22]: This is a really old version. We no longer support this version. In v5 we throw this all the time

**Pull request merged**
